sys/arduino/include/pins_arduino.h

```diff
@@ -0,0 +1,142 @@
+/*
+  pins_arduino.h - Pin definition functions for Arduino
+  Part of Arduino - http://www.arduino.cc/
+
+  Copyright (c) 2007 David A. Mellis
+
+  This library is free software; you can redistribute it and/or
+  modify it under the terms of the GNU Lesser General Public
+  License as published by the Free Software Foundation; either
+  version 2.1 of the License, or (at your option) any later version.
+
+  This library is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+  Lesser General Public License for more details.
+
+  You should have received a copy of the GNU Lesser General
+  Public License along with this library; if not, write to the
+  Free Software Foundation, Inc., 59 Temple Place, Suite 330,
+  Boston, MA  02111-1307  USA
+
+  NOTE: The code in this file comes from https://github.com/arduino/ArduinoCore-avr
+  (see ArduinoCore-avr/variants/.../pins_arduino.h)
+*/
+
+/**
+ * @ingroup    sys_arduino
+ * @{
+ *
+ * @file
+ * @brief      Macros for compatibility with Arduino
+ */
+
+#ifndef PINS_ARDUINO_H
+#define PINS_ARDUINO_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#if defined(BOARD_ARDUINO_LEONARDO)
+
+/**
+ * @brief   Return the PCICR register for the given digital pin
+ */
+#define digitalPinToPCICR(p)    ((((p) >= 8 && (p) <= 11) || ((p) >= 14 && (p) <= 17) || ((p) >= ARDUINO_PIN_A8 && (p) <= ARDUINO_PIN_A10)) ? (&PCICR) : ((uint8_t *)0))
```

Comment:
Would this make sense to you instead?

```suggestion
static inline volatile uint8_t *digitalPinToPCICR(int pin)
{
    if (((pin >= 8) && (pin <= 11))
        || ((pin >= 14) && (pin <= 17))
        || ((pin >= ARDUINO_PIN_A8) && (pin <= ARDUINO_PIN_A10)))
        {
        return CICR;
    }
    return NULL;
}
```

Code style is always a bit of a subjective matter, but I personally have less trouble reading a C function than a long preprocessor macro. Also this would get rid of the "more than 100 chars" warning.

Note that the compiler will be able to evaluate a `static inline` function at compile time if `pin` is known at compile time (assuming that optimizations are enabled), so this doesn't come at performance cost. However, some code might rely on this being a preprocessor macro.

Reply:
This gives the error `‘CICR’ was not declared in this scope; did you mean ‘PCICR’?`
So I will use `&PCICR`
(Otherwise the macros were copy paste from Arduino code.)